src/state/user/reducer.ts

```diff
@@ -137,6 +143,12 @@ const userSlice = createSlice({
       state.timestamp = currentTimestamp()
     },
     setOriginCountry(state, { payload: country }) {
```

Comment:
isn't this slice already persisted through redux-persist?

Reply:
redux should await store rehydration before rendering

Reply:
yeah this is persisted but we really need the info on first render - theoretically we could wait for rehydration before rendering everything but that is really out of scope for this pr

Reply:
we don't wait for rehydration before rendering because it would substantially delay the first frame.